tools/clang/tools/dxclib/dxc.cpp

```diff
@@ -778,7 +778,8 @@ void DxcContext::Recompile(IDxcBlob *pSource, IDxcLibrary *pLibrary,
     IFT(pPdbUtils->GetSourceName(i, &pFileName));
     IFT(pIncludeHandler->insertIncludeFile(pFileName, pSourceFile, 0));
     if (pMainFileName == pFileName) {
-      pCompileSource.Attach(pSourceFile);
+      // Transfer pSourceFile to avoid pPdbUtils release it again.
```

Comment:
Nit: The comment is misleading.

Assignment ('=') would work just fine too, since it would get another ref for pCompileSource so pSourceFile wouldn't release the only ref we hold outside the loop iteration.  This Attach/Detach pattern is just saving one AddRef+Release by taking pSourceFile's reference.

PdbUtils will still release its ref, but we'll have our own valid ref, so it's ok.  In any case, it's not PdbUtils we need to worry about, it's managing our refs properly, so we still have one as long as we need one.

This comment makes it sound like you need to do something special with pSourceFile when retrieved from PdbUtils.  But there's nothing special here.

You could say:
```cpp
// Transfer pSourceFile to avoid extra AddRef+Release.
```

Reply:
Updated in https://github.com/microsoft/DirectXShaderCompiler/pull/5882